src/display/api.js

```diff
@@ -957,10 +957,41 @@ var PDFPageProxy = (function PDFPageProxyClosure() {
      */
     getTextContent: function PDFPageProxy_getTextContent(params) {
       params = params || {};
-      return this.transport.messageHandler.sendWithPromise('GetTextContent', {
+      var readableStream =
+        this.transport.messageHandler.sendWithStream('GetTextContent', {
         pageIndex: this.pageNumber - 1,
         normalizeWhitespace: (params.normalizeWhitespace === true),
         combineTextItems: (params.disableCombineTextItems !== true),
+      }, {
+        highWaterMark: 5,
+        size(textContent) {
+          return textContent.items.length;
+        }
+      });
+      return new Promise((resolve, reject) => {
+        function pump() {
```

Comment:
You can probably simplify this function somewhat, by doing e.g. something like this:
```javascript
function pump() {
  reader.read().then(function({ done, data, }) {
    if (done) {
      resolve(textContent);
    } else {
      for (var key in data.styles) {
        textContent.styles[key] = data.styles[key];
      }
      textContent.items = textContent.items.concat(data.items);
      pump();
    }
  }, reject);
```

(Again, remember that unless you access `this`, there's no point to using an arrow function.)